dipy/nn/finta_model.py

```diff
@@ -0,0 +1,435 @@
+                          name="encoder_conv4",
+                          kernel_initializer=self.conv1d_weights_initializer,
+                          bias_initializer=self.conv1d_biases_initializer)
+        )
+        self.encod_conv5 = pre_pad(
+            layers.Conv1D(512, self.kernel_size, strides=2, padding='valid',
+                          name="encoder_conv5",
+                          kernel_initializer=self.conv1d_weights_initializer,
+                          bias_initializer=self.conv1d_biases_initializer)
+        )
+        self.encod_conv6 = pre_pad(
+            layers.Conv1D(1024, self.kernel_size, strides=1, padding='valid',
+                          name="encoder_conv6",
+                          kernel_initializer=self.conv1d_weights_initializer,
+                          bias_initializer=self.conv1d_biases_initializer)
+        )
+
+        self.flatten = layers.Flatten(name='flatten')
+
+        # For Dense layers
+        # Link: https://pytorch.org/docs/stable/generated/torch.nn.Linear.html (Variables section)
+        # Weights
+        self.k_dense_weights_initializer = np.sqrt(1 / dict_kernel_size_flatten_encoder_shape[self.kernel_size])
+        self.dense_weights_initializer = initializers.RandomUniform(minval=-self.k_dense_weights_initializer,
+                                                                    maxval=self.k_dense_weights_initializer,
+                                                                    seed=2208)
+        # Biases
+        self.k_dense_biases_initializer = self.k_dense_weights_initializer
+        self.dense_biases_initializer = self.dense_weights_initializer
+
+        self.fc1 = layers.Dense(self.latent_space_dims, name='fc1',
+                                kernel_initializer=self.dense_weights_initializer,
+                                bias_initializer=self.dense_biases_initializer)
+
+    def get_config(self):
+        base_config = super().get_config()
+        config = {
+            "latent_space_dims": keras.saving.serialize_keras_object(self.latent_space_dims),
+            "kernel_size": keras.saving.serialize_keras_object(self.kernel_size)
+        }
+        return {**base_config, **config}
+
+    @classmethod
+    def from_config(cls, config):
+        latent_space_dims = keras.saving.deserialize_keras_object(config.pop('latent_space_dims'))
+        kernel_size = keras.saving.deserialize_keras_object(config.pop('kernel_size'))
+        return cls(latent_space_dims, kernel_size, **config)
+
+    def call(self, input_data):
+        """
+        Run the input data through the Encoder.
+        Encode the input_data into a latent
+        space representation.
+
+        Parameters
+        ----------
+        input_data : tf.Tensor or np.ndarray
+            Data to run through the Encoder.
+            Should be of the shape of a standard
+            streamline with 256 points:
+            (n_samples, 256, 3).
```

Comment:
Same comment as above about the hard-coded dimensions and the the need/convenience to use 256-point streamlines. Either here or in the encoder docstring. If 256 is required (e.g. 128 fails to build the AE properly), add a check and test it in the testing.